pyproject.toml

```diff
@@ -77,3 +78,6 @@ ignore_missing_imports = true
 
 [tool.pytest.ini_options]
 norecursedirs = ["bin", "lib", "include", "build"]
+
+[tool.ruff]
+target-version = "py38"
```

Comment:
Our `requires-python` is `>= 3.7`, so this may cause `ruff` to enable fixes/changes that don't work with Python 3.7.

I believe you can drop this config section entirely and `ruff` *should* respect the top-level `requires-python` instead.

(Long term we'll want to remove 3.7 support, but we aim to retain it so long as the current `pip` version supports it.)

```suggestion
```